pkg/git/git.go

```diff
@@ -997,8 +997,25 @@ func (r *Repo) PushToRemote(remote, remoteBranch string) error {
 
 // LsRemote can be used to run `git ls-remote` with the provided args on the
 // repository
-func (r *Repo) LsRemote(args ...string) (string, error) {
-	return r.runGitCmd("ls-remote", args...)
+func (r *Repo) LsRemote(args ...string) (output string, err error) {
+	for i := r.maxRetries + 1; i > 0; i-- {
+		output, err = r.runGitCmd("ls-remote", args...)
+		if err == nil {
+			return output, nil
+		}
+		err = NewNetworkError(err)
+		if !err.(NetworkError).CanRetry() || r.maxRetries == 0 || i == 1 {
+			return "", err
+		}
+
+		waitTime := math.Pow(2, float64(r.maxRetries-i))
+		logrus.Errorf(
+			"Error executing ls-remote (will retry %d more times in %.0f secs): %s",
```

Comment:
```suggestion
			"Executing ls-remote (will retry %d more times in %.0f secs): %s",
```
The log level already indicates that it's an error.